src/commands/ws/run.js

```diff
@@ -2,15 +2,17 @@
 import type {Args, Opts} from '../../types';
 import Project from '../../Project';
 import * as yarn from '../../utils/yarn';
+import filterWorkspaces from '../../utils/filterWorkspaces';
 
 export default async function run(args: Args, opts: Opts) {
   let cwd = process.cwd();
   let project = await Project.init(cwd);
   let workspaces = await project.getWorkspaces();
+  let filteredWorkspaces = filterWorkspaces(workspaces, opts);
```

Comment:
I'd really prefer if we manually specified which options got passed between functions everywhere. I don't like passing unrelated objects around.

In general I think you should treat all "options" or "config" in Flow as ["Exact types"](https://flow.org/en/docs/types/objects/#toc-exact-object-types)

```js
function fn1(opts1: {| foo: string, bar: string |}) {
  fn2(opts1); // Error: `bar` is not allowed in type of `opts2`
}

function fn2(opts2: {| foo: string |}) {
  // ...
}
```

Instead, if you specify the options manually, it will work:

```js
fn2({ foo: opts.foo }); // Works
```

Reply:
Yup, for sure